lib/matplotlib/tests/test_colors.py

```diff
@@ -265,6 +265,86 @@ def test_BoundaryNorm():
     vals = np.ma.masked_invalid([np.Inf])
     assert np.all(bn(vals).mask)
 
+    # Testing extend keyword
+    bounds = [1, 2, 3]
+    cmap = cm.get_cmap('jet')
+
+    refnorm = mcolors.BoundaryNorm([0] + bounds + [4], cmap.N)
+    mynorm = mcolors.BoundaryNorm(bounds, cmap.N, extend='both')
+    x = np.random.random(100) + 1.5
+    np.testing.assert_array_equal(refnorm(x), mynorm(x))
+
+    # Min and max
+    cmref = mcolors.ListedColormap(['blue', 'red'])
+    cmref.set_over('black')
+    cmref.set_under('white')
+
+    cmshould = mcolors.ListedColormap(['white', 'blue', 'red', 'black'])
+    cmshould.set_over(cmshould(cmshould.N))
+    cmshould.set_under(cmshould(0))
+
+    refnorm = mcolors.BoundaryNorm(bounds, cmref.N)
+    mynorm = mcolors.BoundaryNorm(bounds, cmshould.N, extend='both')
+    np.testing.assert_array_equal(refnorm.vmin, mynorm.vmin)
```

Comment:
many of these could just be plain asserts

Reply:
I considerably reworked the tests.